java/src/main/java/ai/rapids/cudf/CudaException.java

```diff
@@ -28,11 +28,150 @@
  * don't switch between threads for different parts of processing that can be retried as a chunk.
  */
 public class CudaException extends RuntimeException {
-  CudaException(String message) {
+  CudaException(String message, String cudaErrorName) {
     super(message);
+    cudaError = CudaError.valueOf(cudaErrorName);
```

Comment:
I'm not a fan of using `valueOf` here.  It is scraping string messages to discern meaning again.  It also has the significant problem that f there ever was a string message that is not in the expected list, trying to throw a CudaException will instead throw an `IllegalArgumentException`, eclipsing the original exception we're trying to throw.

I recommend passing the original CUDA error value here, not the stringified form of it, and have Java try to map that to the enum values using the native IDs, if we even keep the enum values around.  I could see just having this API be the exception message and an accessor method to get the integral value of the CUDA error.  If we do keep the enum values around, then there would need to be an enum value that represents "Java doesn't have a representation of this native CUDA error type."

Reply:
I replaced the error name with the error code.